avm/res/recovery-services/vault/main.bicep

```
@@ -0,0 +1,458 @@
+metadata name = 'Recovery Services Vaults'
+metadata description = 'This module deploys a Recovery Services Vault.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Azure Recovery Service Vault.')
+param name string
+
+@description('Optional. The storage configuration for the Azure Recovery Service Vault.')
+param backupStorageConfig object = {}
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. List of all backup policies.')
+param backupPolicies array = []
+
+@description('Optional. The backup configuration.')
+param backupConfig object = {}
+
+@description('Optional. List of all protection containers.')
+@minLength(0)
+param protectionContainers array = []
+
+@description('Optional. List of all replication fabrics.')
+@minLength(0)
+param replicationFabrics array = []
+
+@description('Optional. List of all replication policies.')
+@minLength(0)
+param replicationPolicies array = []
+
+@description('Optional. Replication alert settings.')
+param replicationAlertSettings object = {}
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Tags of the Recovery Service Vault resource.')
+param tags object?
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints privateEndpointType
+
+@description('Optional. Monitoring Settings of the vault.')
+param monitoringSettings object = {}
+
+@description('Optional. Security Settings of the vault.')
+param securitySettings object = {}
+
+@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Disabled'
+
+var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities) ? {
+  type: (managedIdentities.?systemAssigned ?? false) ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : null)
```

```suggestion
  type: (managedIdentities.?systemAssigned ?? false) ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : 'None')
```

This throws a warning, above suggestion should help.